hip-service/Link/Patient/LinkPatient.cs

```diff
@@ -0,0 +1,122 @@
+        {
+
+            var requestPatient = request.Patient;
+            var patient = _patientRepository.GetPatientInfoWithReferenceNumber(requestPatient.ReferenceNumber);
+            
+            if (patient == null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, ErrorResponse>
+                    (null, new ErrorResponse(new Error(ErrorCode.NoPatientFound, "No patient found")));
+            }
+            var programs = new List<hip_service.Discovery.Patient.models.CareContext>();
+            foreach (var careContext in requestPatient.CareContexts)
+            {
+                if (_patientRepository.GetProgramInfo(requestPatient.ReferenceNumber,
+                        careContext.ReferenceNumber) != null)
+                {
+                    programs.Add(_patientRepository.GetProgramInfo(requestPatient.ReferenceNumber,
+                        careContext.ReferenceNumber));
+                }
+            }
+
+            if (programs.Count != requestPatient.CareContexts.Count())
+            {
+                return new Tuple<PatientLinkReferenceResponse, ErrorResponse>
+                    (null, new ErrorResponse(new Error(ErrorCode.CareContextNotFound, "Care context not found for given patient")));
+            }           
+            
+            var linkRefNumber = Guid.NewGuid().ToString();
+            
+            // method call for generating the OTP
+            var session = new Session(linkRefNumber, new Communication(IdentifierType.MOBILE,patient.PhoneNumber));
+
+            if (await _patientVerification.GenerateVerificationToken(session) != null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, ErrorResponse>
+                    (null, new ErrorResponse(new Error(ErrorCode.OtpInValid, "Unable to create token")));
+            }
+            
+            var (_, exception) = await linkPatientRepository.SaveLinkPatientDetails(linkRefNumber, requestPatient.ConsentManagerId,
+                requestPatient.ConsentManagerUserId, requestPatient.ReferenceNumber, requestPatient.CareContexts
+                    .Select(context => context.ReferenceNumber).ToArray());
+
+            if (exception != null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, ErrorResponse>
+                    (null, new ErrorResponse(new Error(ErrorCode.CareContextNotFound, "Unable to store data to Database")));
+            }
+
+            const string dateTimeFormat = "yyyy-MM-ddTHH:mm:ssZ";
+            var expiry = new DateTime (DateTime.Today.Year, DateTime.Today.Month, DateTime.Today.Day, 
+                    DateTime.Now.Hour, DateTime.Now.Minute+1, DateTime.Now.Second).ToUniversalTime().
+                ToString(dateTimeFormat); 
+
+            var meta = new LinkReferenceMeta(nameof(IdentifierType.MOBILE).ToUpper(), 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4769144)

Reply:
Fixed

Reply:
Done